### ISSUE TITLE
Translation

### ISSUE THREAD

**Issue by GoogleCodeExporter** [alistairreilly/andors-trail#146]
```
I think that it is necessary to begin the process of translating the game into 
different languages. Can organize this process? I think the community will help 
;)
```

Original issue reported on code.google.com by `gVz...@gmail.com` on 27 Jan 2011 at 4:22

**Comment by GoogleCodeExporter** [comment]
```
Sure, I'm willing to translate into dutch
```

Original comment by `dry...@gmail.com` on 28 Jan 2011 at 8:54

**Comment by GoogleCodeExporter** [comment]
```
Thanks for the report.

Good idea. This is something that I would really like to see happen, it would 
make the game more accessible to a wider gaming audience. I have received 
emails from people wanting to do the Italian and Russian translation as well. 

I could probably to the Swedish translation also :)

I think there are a couple of parts that are not prepared to be translated. 
Monster and item names might be a problem at the moment, but dialog, quest log 
and the whole game interface is prepared for translation. I should probably 
write some wiki page to clarify what parts that are required to do a 
translation. And I should probably to the required coding to allow item and 
monster translations as well.

Let's see if I find the time to write that wiki page. In the meantime, I'll 
point you to the right direction as to where to start translating: All these 
files would need to be translated:
http://code.google.com/p/andors-trail/source/browse/trunk/AndorsTrail/res/values
/

Thanks again for the report and for wanting to contribute to the game!
```

Original comment by `oskar.wi...@gmail.com` on 29 Jan 2011 at 5:13
- Added labels: **Type-Enhancement**
- Removed labels: **Type-Defect**

**Comment by GoogleCodeExporter** [comment]
```
Hungarian is OK for me.
```

Original comment by `surrano` on 31 Jan 2011 at 9:32

**Comment by GoogleCodeExporter** [comment]
```
[deleted comment]
```

**Comment by GoogleCodeExporter** [comment]
```
Need add strings for settings menu

```

Original comment by `D.Dmitri...@gmail.com` on 31 Jan 2011 at 9:39

**Comment by GoogleCodeExporter** [comment]
```
BTW how about international characters? What is the proposed code page / 
unicode format you recommend?
```

Original comment by `surrano` on 31 Jan 2011 at 9:42

**Comment by GoogleCodeExporter** [comment]
```
[deleted comment]
```

**Comment by GoogleCodeExporter** [comment]
```
UTF-8
```

Original comment by `D.Dmitri...@gmail.com` on 31 Jan 2011 at 6:16

**Comment by GoogleCodeExporter** [comment]
```
Yes, utf-8 is what is used now. Or, to be more precise - the codepage specified 
by the xml doctype tag in each file.
```

Original comment by `oskar.wi...@gmail.com` on 31 Jan 2011 at 6:33

**Comment by GoogleCodeExporter** [comment]
```
Should've figured it out... 
*Surranó bonks himself on the head*
```

Original comment by `surrano` on 31 Jan 2011 at 6:35

**Comment by GoogleCodeExporter** [comment]
```
I have started investigating this more closely. Translation of the 
conversationlist, quest log and strings.xml (interface data) will be no 
problem, they are ready to translate as is. Nor will there be a problem 
translating item names, they are persisted by the internal tag instead of the 
name.

However, there is a problem translating monster names currently. Savegames 
contain the monster name to specify what monster is placed where on the map. 
This is something I really should have thought of from the beginning. Me being 
a database guy at work, I am really ashamed of this :) Anyway, that's the way 
savegames are persisted now. If we do some translation of the monster names, 
then savegames made by 0.6.7 (or earlier) will not load into 0.6.8 . Let's 
figure out some way do translate this nicely into translated names, and I'll 
modify the savegame creation to store some language-independent identifier for 
the monsters.
```

Original comment by `oskar.wi...@gmail.com` on 31 Jan 2011 at 6:40

**Comment by GoogleCodeExporter** [comment]
```
Additionally, the text displayed on the settings screen is currently only 
implemented in res/xml/preferences.xml . I'll move those strings into 
res/values/strings.xml for 0.6.8 .
```

Original comment by `oskar.wi...@gmail.com` on 31 Jan 2011 at 6:46

**Comment by GoogleCodeExporter** [comment]
```
Another thing that cannot be translated right now: All map signs are currently 
located as strings inside the map file. These will also have to be moved into 
some language-dependent resource file.

I'm thinking we change the map signs into re-using the conversation code. So 
stepping on a map sign would have the same effect as opening up a conversation.
```

Original comment by `oskar.wi...@gmail.com` on 31 Jan 2011 at 6:50

**Comment by GoogleCodeExporter** [comment]
```
Thank you give  attention  to  the localization
In files what i send you today none quest monsters allready translated

/Dreamer...
```

Original comment by `D.Dmitri...@gmail.com` on 31 Jan 2011 at 7:00

**Comment by GoogleCodeExporter** [comment]
```
Wouldn't it also work if you save English monster names in the savegame?
You may get the IDs when loading and display the real names when attacking.
```

Original comment by `SamuelPl...@gmail.com` on 31 Jan 2011 at 7:20

**Comment by GoogleCodeExporter** [comment]
```
Sorry, I'm still not clear on this... how do you want us to do the translation? 
Just give you a conversationlist_<language>.xml? How will you handle newly 
added and/or changed conversations?
```

Original comment by `dry...@gmail.com` on 3 Feb 2011 at 9:22

**Comment by GoogleCodeExporter** [comment]
```
I think, Oscar notify us about newest or changed conversations ;)
```

Original comment by `D.Dmitri...@gmail.com` on 3 Feb 2011 at 5:48

**Comment by GoogleCodeExporter** [comment]
```
Ok but how do we deliver the translation? Really as a patch? I can't imagine 
that...
```

Original comment by `dry...@gmail.com` on 8 Feb 2011 at 12:10

**Comment by GoogleCodeExporter** [comment]
```
You can email them to Oskar.
```

Original comment by `sdeva...@gmail.com` on 8 Feb 2011 at 8:57

**Comment by GoogleCodeExporter** [comment]
```
I have now also updated the codebase for v0.6.8 to support translation of 
preferences and map signs. 

Map signs now use the same interface as conversations. A really nice 
side-effect of this is that map signs can now potentially contain scriptable 
tasks like the usual conversation options. So stepping on a certain tile could 
give different messages depending on the state of some quest.
```

Original comment by `oskar.wi...@gmail.com` on 10 Feb 2011 at 8:11

**Comment by GoogleCodeExporter** [comment]
```
wow... something like... contents of a "Wanted" poster, by any chance? :)
```

Original comment by `surrano` on 11 Feb 2011 at 11:09

**Comment by GoogleCodeExporter** [comment]
```
Amazing!
Where I can download changed strings for translate?
```

Original comment by `D.Dmitri...@gmail.com` on 11 Feb 2011 at 9:39

**Comment by GoogleCodeExporter** [comment]
```
I would love to do a Polish translation? What should I do to do it?
```

Original comment by `arng...@gmail.com` on 20 Apr 2011 at 8:05

**Comment by GoogleCodeExporter** [comment]
Original comment by `oskar.wi...@gmail.com` on 4 May 2011 at 5:41
- Added labels: **Type-Translation**
- Removed labels: **Type-Enhancement**

**Comment by GoogleCodeExporter** [comment]
```
I wrote a wiki page about what needs to be done in a translation:
http://code.google.com/p/andors-trail/wiki/Translation_Howto

Should we add something more to that wiki page?

Also, is there anything left in this issue or should we close it? "Translation" 
will be a continuous process that may never be "finished".
```

Original comment by `oskar.wi...@gmail.com` on 9 May 2011 at 5:23

**Comment by GoogleCodeExporter** [comment]
```
Marking this issue as ready for v0.6.9, with the excellent efforts of the 
translators. "Translation" as a general term will never be completed, but a 
continuous process, as previously stated.

Further translation efforts can be found in issue 194.
```

Original comment by `oskar.wi...@gmail.com` on 16 May 2011 at 6:34
- Changed state: **Ready**
- Added labels: **Milestone-0.6.9**

**Comment by GoogleCodeExporter** [comment]
```
Released in v0.6.9 . Closing.
```

Original comment by `oskar.wi...@gmail.com` on 29 Jun 2011 at 7:50
- Changed state: **Done**
